scripts/tests/chiptest/linux.py

```diff
@@ -182,35 +188,36 @@ def _setup_app_link_up(self, wait_for_dad=True):
         for command in self.COMMANDS_APP_LINK_UP:
             self._run(command)
         if wait_for_dad:
-            self._wait_for_duplicate_address_detection()
+            self.wait_for_duplicate_address_detection()
 
     def _setup_tool_link_up(self, wait_for_dad=True):
         for command in self.COMMANDS_TOOL_LINK_UP:
             self._run(command)
         if wait_for_dad:
-            self._wait_for_duplicate_address_detection()
-
-    def _run(self, command: str):
-        command = command.format(app_link_name=self.app_link_name,
-                                 tool_link_name=self.tool_link_name,
-                                 index=self.index)
-        log.debug("Executing: '%s'", command)
-        if subprocess.run(command.split()).returncode != 0:
-            log.error("Failed to execute '%s'", command)
-            log.error("Are you using --privileged if running in docker?")
-            sys.exit(1)
+            self.wait_for_duplicate_address_detection()
+
+    def _run(self, *command: str):
+        for c in command:
+            c = c.format(app_link_name=self.app_link_name, tool_link_name=self.tool_link_name, index=self.index)
+            log.debug("Executing: '%s'", c)
+            if subprocess.run(c.split()).returncode != 0:
+                raise RuntimeError(f"Failed to execute '{c}'. Are you using --privileged if running in docker?")
 
     def terminate(self):
-        for command in self.COMMANDS_TERMINATE:
-            self._run(command)
+        """Execute all down commands in reverse order gracefully omitting errors."""
+        for cmd in self.COMMANDS_TERMINATE:
+            try:
+                self._run(cmd)
+            except Exception as e:
+                log.warning("Encountered error during namespace termination: %s", e, exc_info=True)
```

Comment:
Do we really need stack trace here? Also, in other similar place you are logging exceptions as `error` while here as `warning`.

Reply:
Seeing a call trace on terminal triggers "there is a bug" reaction (at least that's my case). If such trace is not a bug but simply some runtime error which should be handled "somehow", by showing traces in random places we are training ourselves to treat traces as a noise, which is not good in my opinion.

Reply:
If the program can recover form here onward, then it's a warning, if it cannot recover (functionality will be degraded) then it's an error, if it can not function correctly, then it's a critical error. That's how I see it. So, in this particular case it seems like a warning - it's a termination logic anyway... The same applies to the `def cleanup() -> None` in the test suite code, IMHO.